engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/serialization/ConfluentKafkaAvroDeserializationSchemaFactory.scala

```diff
@@ -4,36 +4,68 @@ import io.confluent.kafka.serializers._
 import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.kafka.common.serialization.Deserializer
 import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.client.ConfluentSchemaRegistryClientFactory
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.ConfluentKafkaAvroDeserializationMode.ConfluentKafkaAvroDeserializationMode
+import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.serialization.serializer.{ConfluentKafkaAvroDeserializer, ConfluentStaticKafkaAvroDeserializer}
 import pl.touk.nussknacker.engine.kafka.KafkaConfig
-import pl.touk.nussknacker.engine.kafka.serialization.{KafkaVersionAwareValueDeserializationSchemaFactory, KafkaVersionAwareKeyValueDeserializationSchemaFactory}
+import pl.touk.nussknacker.engine.kafka.serialization.{KafkaVersionAwareKeyValueDeserializationSchemaFactory, KafkaVersionAwareValueDeserializationSchemaFactory}
 
 trait ConfluentKafkaAvroDeserializerFactory {
   import collection.JavaConverters._
 
-  protected def createDeserializer[T](schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory, kafkaConfig: KafkaConfig, isKey: Boolean, useSpecificAvroReader: Boolean): Deserializer[T] = {
+  protected def createDeserializer[T](mode: ConfluentKafkaAvroDeserializationMode,
+                                      topic: String,
+                                      version: Option[Int],
+                                      schemaRegistryClientFactory: ConfluentSchemaRegistryClientFactory,
+                                      kafkaConfig: KafkaConfig,
+                                      isKey: Boolean,
+                                      useSpecificAvroReader: Boolean): Deserializer[T] = {
     val schemaRegistryClient = schemaRegistryClientFactory.createSchemaRegistryClient(kafkaConfig)
-    val deserializer = new KafkaAvroDeserializer(schemaRegistryClient.client)
+
+    val deserializer = mode match {
+      case ConfluentKafkaAvroDeserializationMode.STATIC =>
```

Comment:
Can we add some description/test how it will behave on compatible/not-compatible schemas?

Reply:
It makes sens ;)